utils/defillama.ts

```diff
@@ -1,68 +1,42 @@
 import axios from "axios";
 
-const baseUrl = "https://api.llama.fi/protocol/uma-protocol";
-const defillamaUrls: string[] = [
-  "https://api.llama.fi/protocol/polymarket",
-  "https://api.llama.fi/protocol/sherlock",
-  "https://api.llama.fi/protocol/jarvis-network",
-  "https://api.llama.fi/protocol/across",
-  "https://api.llama.fi/protocol/uma-protocol",
-];
+const baseUrl = "https://api.llama.fi/oracles";
+
+interface TvsData {
+  tvl: number;
+}
 
 interface TvlData {
   date: string;
   totalLiquidityUSD: number;
 }
 
-export async function getTvl(url: string = baseUrl) {
+export async function getDefillamaTvl(url: string = baseUrl) {
   const { data } = await axios.get(url);
-  const defillamaTvlData: TvlData[] = data.tvl.slice(-1);
-  const latestTvl = defillamaTvlData[0].totalLiquidityUSD;
-  return latestTvl;
-}
+  const defillamaTvlData: TvsData =
+    data.chart[Object.keys(data.chart)[Object.keys(data.chart).length - 1]][
+      "UMA"
+    ];
+  const latestTvl = defillamaTvlData.tvl;
 
-export async function getPercentChange(url: string = baseUrl) {
-  const { data } = await axios.get(url);
-  const defillamaTvlData: TvlData[] = data.tvl.slice(-2);
-  const latestTvl = defillamaTvlData[1].totalLiquidityUSD;
-  const previousDayTvl = defillamaTvlData[0].totalLiquidityUSD;
-  return {
-    latestTvl,
-    previousDayTvl,
-  };
-}
-
-export async function getDefillamaTvl(url: string[] = defillamaUrls) {
-  const totalLatestTvl = [];
-
-  for (let i = 0; i < defillamaUrls.length; i++) {
-    const todayTvl = await getTvl(url[i]);
-    totalLatestTvl.push(todayTvl);
-  }
-  const latestTvl = totalLatestTvl.reduce(function (totalLatestTvl, a) {
-    return totalLatestTvl + a;
-  }, 0);
   return latestTvl;
 }
 
-export async function getDefillamaPercentChange(url: string[] = defillamaUrls) {
-  const todayTvl = [];
-  const yesterdayTvl = [];
+export async function getDefillamaPercentChange(url: string = baseUrl) {
+  const { data } = await axios.get(url);
+  const latestTvl: TvsData =
+    data.chart[Object.keys(data.chart)[Object.keys(data.chart).length - 1]][
```

Comment:
same with these, not quite sure what these are doing but it looks like you could save the result in a var rather than keep recalculating these indexes

Reply:
I made an update attempting to reduce complexity and avoid recalculating the indexes. Let me know if this is not what you were looking for.